activesupport/lib/active_support/evented_file_update_checker.rb

```diff
@@ -28,11 +57,12 @@ def initialize(files, dirs = {}, &block)
             raise LoadError, "Could not load the 'listen' gem. Add `gem 'listen'` to the development group of your Gemfile", e.backtrace
           end
         end
-        Listen.to(*dtw, &method(:changed)).start
       end
+      boot!
     end
 
     def updated?
+      boot! unless @pid_hash[Process.pid]
```

Comment:
We should still use an atomic even if it's not a hash. While it's technically fine to have more than one listener, it's unnecessary and easy to avoid.